### ISSUE TITLE
Mask from Subject Selection

### ISSUE THREAD

**Issue by tobiasBora** [Beep6581/RawTherapee#7537]
Is is often really helpful to select a subject precisely (to apply an effect only to the subject, or to remove the subject from the another mask etc), and is done in so many lightroom tutorials that I feel a bit lost when using any open-source software (see e.g. https://www.youtube.com/watch?v=FMaB03h1rQk). Would it be possible to add an AI-masking tool in RT? Some possible implementation details are discussed in darktable (https://github.com/darktable-org/darktable/issues/12295) that mention, e.g., Segment Anything https://segment-anything.com/, https://github.com/danielgatis/rembg , SAM2, FAST-SAM, Mobile SAM…

**Comment by digitalcarp** [comment]
I've experimented with the open source models out there before. You can get very good AI masking on a lot of tasks using them with an ONNX runtime.

SAM-HQ, MobileSAM, and SAM2 gets u pretty good semantic segmentation to mask user selections. Add GroundingDINO to get text based segmentation. 

BiRefNet/U2Net can be used for foreground/background masking.

YuNet face detection + some face parsing models can usually isolate facial features well.

A combination of image matting models (e.g. ViTMatte) with CV techniques like erosion and dilation can be used to enhance or fine-tune results from the previously mentioned models.

There's also models for depth estimation like DepthAnythingV2.

Some issues to consider when using open source models:

- Mostly trained on 1024px images or smaller, need to partition ur image or scale inputs and outputs appropriately
- almost exclusively trained on SRGB images so you need to decide what image u export to the model
- Make sure licensing is okay for use
  - some models r non commercials use or academic use only
  - some pretrained models have similar licensing restrictions
- Make the AI part an optional add on as onnxruntime is a large file that not everyone wants to use

For RawTherapee, any masking improvements would be good as there isn't even brush masking yet. Easiest path forward would probably be an entirely new Masking UI that doesn't have to wrangle with the existing canvas GUI code.

Anyways those r my thoughts and experiments from several months ago before Papers with Code got shut down :(

**Comment by tobiasBora** [comment]
An intermediate solution could also to provide raster masks and create it from an external app like darktable/art.
